web3/auto/infura/__init__.py

```diff
@@ -1,13 +1,13 @@
-from web3 import Web3
-from web3.providers.auto import (
-    load_provider_from_uri,
-)
+import logging
 
-from .endpoints import (
-    INFURA_MAINNET_DOMAIN,
-    build_infura_url,
-)
+from .mainnet import w3  # noqa: F401
 
-_infura_url = build_infura_url(INFURA_MAINNET_DOMAIN)
 
-w3 = Web3(load_provider_from_uri(_infura_url))
+logging.getLogger('web3.auto.infura').warning(
```

Comment:
Maybe we should just remove the default.  I'm not sure a warning will really do enough if we think this is an issue (which I tend to agree that it is).  Having explicit imports indicate what network makes sense to me.

Reply:
You could also emit to `stderr` which is probably what you need to do if you *really* want to make sure people see it, since a lot of people don't see python warnings.

Reply:
I just recommitted the old `__init__.py`. Something is really fragile in that test suite, and it's not worth it to debug.